### DISCUSSION TITLE
Option to manually edit a pinned tab

### DISCUSSION THREAD

**Discussion by rcsaquino** [zen-browser/desktop#2221] [Ideas]
A feature to manually edit pinned tabs would give users more control and solve common issues with dynamic URLs. Here's why it's important:

1. **Prevent unwanted redirects:** Some sites, like Facebook Messenger, automatically redirect to your last chat or specific content. Pinning that tab locks in the redirected URL. With manual editing, you could pin the homepage instead, ensuring it always opens fresh.

2. **Avoid dynamic URLs:** Sites that update URLs daily (e.g., news sites) often redirects to the today's page (e.g., somewebsite.com/news-for-Oct-18-2024). Editing the URL lets you pin the base site (somewebsite.com) to get the latest content every time.

3. **Optional title editing:** Sites like YouTube add notification counts to the tab title. Editing the title manually would let you clean it up for a neater interface.

This feature would eliminate workarounds and improve user experience by making pinned tabs more flexible and customizable.

**Comment by stripedgoat** [comment]


**Reply by rcsaquino** [reply]
Read 1 and 2.

As an example, try pinning `facebook.com`. Not `web.facebook.com` or `facebook.com/some-stuff-here`. Just pure `facebook.com`. You'll find that it's hard because it will redirect before you could pin it.

**Comment by pressRtowin** [comment]
This is an essential feature that must be added.

**Comment by pressRtowin** [comment]
Anyone know of a workaround in the meantime?

**Reply by loofkid** [reply]
It's not a user-friendly way to do it (and no guarantee it will always work), but I was able to edit the zen-sessions.jsonlz4 file to update the urls.

Essentially: 

1. Find your user profile in about:support, find the zen-sessions.jsonlz4 file.
2. Fully close Zen.
3. Use mozlz4 (https://github.com/jusw85/mozlz4 - I am on a Mac, so I had to build it from source) to decompress the file to json (`mozlz4 -x {{path to zen-sessions.jsonlz4}} {{path to zen-sessions.jsonlz4}}.json`).
4. Open the zen-sessions.jsonlz4.json file and find and replace the URLs/pinned tabs you want to update. Save.
5. Recompress the jsonlz4.json file, overwriting the existing one (`mozlz4 -z {{path to zen-sessions.jsonlz4}}.json {{path to zen-sessions.jsonlz4}}`).
6. Reopen Zen. Your tabs should be updated.

I agree that this should be implemented. Kinda odd that it's been sitting out here for a year and a half with nothing, when it does seem (from a user point of view) like it would be an easy add.

**Comment by TheScooterKing** [comment]
It would be just as important to edit unpinned tabs. It really helps with my Google Fu, among other things

**Comment by smidge84** [comment]
I would appreciate this feature too. Was useful in the Arc Browser.

**Comment by askreet** [comment]
I'm finding it hard to set up Zen because this is missing. I have authentication URLs for things like AWS SAML that redirect to generic login portals, but I need to start at the specific URL.

**Comment by NickBeukema** [comment]
I loved this feature in Arc, definitely would be useful to have here in Zen.

In the meantime, a workaround is opening a new tab (say to google.com), visit the Network panel in the developer tools (F12), find the dropdown that says "No Throttling" and choose Offline. Then visit the exact URL you're hoping to pin. This will block the redirect and allow you to pin it exact. Turn off the throttling and refresh the page, this should setup the pin correctly.

Note, if you've already cached the redirect, you may need to open it in a different container and then move it to the space you want it in.

**Comment by TomLisankie** [comment]
I'd really like this feature. Should be a quick fix.

**Comment by koichiio** [comment]
For anyone who doesn't know, this feature was [published](https://zen-browser.app/release-notes/#:~:text=152%2E0%2E3-,Added,URL) in June, 2026.
